language/module.md

```diff
@@ -0,0 +1,70 @@
+# Puppet Module Schema
+
+This document describes allowed (optional) and required files and directories in
+a Puppet Module.
+
+## What is a Puppet Module
+
+A module in the sense of this document is a compressed tar archive. It is
+usually distributed via [forge.puppet.com](https://forge.puppet.com/). A module
+is usually developed in a git repository. Some files in the git repository are
+only used for development and testing. They shall not be released.
```

Comment:
If we interpret `shall` using RFC2119 then this is a prohibition. I'd not go as far.
```suggestion
only used for development and testing. They should not be released.
```

My reasoning is that on the server side you will still have old modules with those assets present, so you need to deal with them anyway.

One reason to include them is to provide everything needed to make offline changes, including tests (provided you have the other modules and dependencies too). In a distro like Fedora this is what happens for gems: they should run the test suite of that gem to verify it's working. Now granted, few Puppet modules are being packaged but it is a valid use case.

Reply:
but do you package the tests on fedora? on arch, we run available tests for gems during the packaging, but the tests are usually not part of the package.